src/google/protobuf/compiler/objectivec/objectivec_helpers.cc

```diff
@@ -839,15 +840,30 @@ string BuildCommentsString(const SourceLocation& location) {
   while (!lines.empty() && lines.back().empty()) {
     lines.pop_back();
   }
-  string prefix("///");
-  string suffix("\n");
+
+  string prefix;
+  string suffix;
   string final_comments;
+  string epilogue;
+
+  if (prefer_single_line && lines.size() == 1) {
+    prefix = "/** ";
+    suffix = " */\n";
+  } else {
+    prefix = " * ";
+    suffix = "\n";
+    final_comments += "/**\n";
+    epilogue = " **/\n";
+  }
+
   for (int i = 0; i < lines.size(); i++) {
     // HeaderDoc uses '\' and '@' for markers; escape them.
-    const string line = StringReplace(lines[i], "\\", "\\\\", true);
+    const string line = StripPrefixString(
+        StringReplace(lines[i], "\\", "\\\\", true), " ");
```

Comment:
Can you add some comments to the objc test protos to include things that this escaping code will trigger, and then inspect the results, just to make sure they are complete?  I'm especially worried about stuff like: `image/*`, `*/plain` (because they look like markup for C comments); as well as directives (or things that look like directives: `@param`, `@c`, `@me`

(you'll want to check how they do in Xcode from it parsing the headers as well as how they do in generated docs)